sample/src/androidTest/java/com/schibsted/spain/barista/sample/util/SpyFailureHandlerRule.kt

```diff
@@ -0,0 +1,47 @@
+package com.schibsted.spain.barista.sample.util
+
+import android.support.test.espresso.Espresso
+import com.schibsted.spain.barista.internal.failurehandler.SpyFailureHandler
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+class SpyFailureHandlerRule : TestRule {
+
+  private val failureHandler = SpyFailureHandler()
+  private var failed = false
+  private var asserted = false
+
+  override fun apply(base: Statement, description: Description): Statement {
+
+    return object : Statement() {
+      override fun evaluate() {
+        Espresso.setFailureHandler(failureHandler)
+        try {
+          base.evaluate()
+        } catch (t: Throwable) {
+          failed = true
+          throw t
+        } finally {
+          if (!failed && !asserted) {
```

Comment:
Meeeh...

Reply:
I **don't** like this code at all... I know I should make a constructive criticism and give you an alternative but I don't have it so... Just I wanted to let you know that I'm not happy with it 🤷‍♂️

Reply:
Then shut up!

Just kidding, if your _spider sense_ is telling you this is bad, please think about what you don't like and how to improve it.
Is it the behavior of the code that you don't like? Or how the implementation looks?
I'm absolutely open to suggestion!

Reply:
It's the implementation, IMO using flags to control the flow of the code is not a good practice because it's too weak and could end up in an unexpected behaviour.
On the other hand, this is not a key code neither complex and I'm sure it won't fail... In addition it's a test code so, what we should worry about it?
Go ahead and forget my mimimi's!

Reply:
Done!